main.go

```diff
@@ -1,8 +1,8 @@
 package main
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/plugin"
-	"github.com/terraform-providers/terraform-provider-null/null"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+	"github.com/terraform-providers/terraform-provider-null/internal/provider"
 )
 
 func main() {
```

Comment:
this seems like it wouldn't compile below (`null.Provider`)

Reply:
It compiles because the package name is `null`, which does not match the import path. Happy to switch provider package names to `package provider` throughout (as in scaffolding) if you think this is a good convention.